### PR TITLE
Updated setup.py to exclude tests

### PR DESCRIPTION
**Pull request by TechnicalPirate**
# Problem #
The `.whl` available on PyPi for this repository contains its entire `tests` submodule. 

The total size of the `pygithub` module is less than `1mb` without tests and `>23mb` with. 

Previous discussions on this topic indicated the removal of tests would be accepted if someone created a pull request to do so: https://github.com/PyGithub/PyGithub/issues/672 

This is that pull request! 😄

# Solution/Implementation # 

- Moved "tests" to root of the repository ( outside the package ) 
    - Added `tests.` module path to the imports in `AllTests.py`
- Updated `setup.py`
    - Removed `package_data` from product
    - Removed "tests" from packages definition
    - Repointed `test_suite` to new location
- Updated `travis.yml`
    - Added conditional ( if not `py2.7` ) conversion of code to `py3`

_... apologies for all the failed build spam below - new to working with Travis CI / had some gotcha moments!_

### CODE REVIEW

**Comment by sfdye** [comment]
Thanks for this! It seems there are some conflicts so I can't merge this.

**Comment by TechnicalPirate** [comment]
Hi!

I just checked, i don't see any new pushes to master since the last build
of this PR which passed :
https://travis-ci.org/PyGithub/PyGithub/builds/487063222?utm_source=github_status&utm_medium=notification

What conflicts are you seeing?

:)

**Comment by sfdye** [comment]
image

This is what I see

**Comment by TechnicalPirate** [comment]
@sfdye - I had the wrong remote set for pulling :) - Fixed, repulled, re-merged - new commit building... if it passes hopefully those conflicts are gone!

**Comment by TechnicalPirate** [comment]
I see there was another change. I've merged this onto this PR.

**Comment by TechnicalPirate** [comment]
I've updated this PR once again @sfdye :) 

I'll keep maintaining it until the end of this week, but if we cannot merge it soon I will stop as its costing me development time on other tasks!

File: tests/ReplayData/ConditionalRequestUpdate.testUpdateObjectWithoutEtag.txt [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-{"clone_url":"https://github.com/jacquev6/PyGithub.git","has_downloads":true,"watchers":15,"git_url":"git://github.com/jacquev6/PyGithub.git","updated_at":"2012-05-27T06:55:28Z","permissions":{"pull":true,"admin":true,"push":true},"homepage":"http://vincent-jacques.net/PyGithub","url":"https://api.github.com/repos/jacquev6/PyGithub","mirror_url":null,"has_wiki":false,"has_issues":true,"fork":false,"forks":3,"size":308,"private":false,"open_issues":16,"svn_url":"https://github.com/jacquev6/PyGithub","owner":{"url":"https://api.github.com/users/jacquev6","avatar_url":"https://secure.gravatar.com/avatar/b68de5ae38616c296fa345d2b9df2225?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-140.png","gravatar_id":"b68de5ae38616c296fa345d2b9df2225","login":"jacquev6","id":327146},"name":"PyGithub","language":"Python","description":"Python library implementing the full Github API v3","ssh_url":"git@github.com:jacquev6/PyGithub.git","pushed_at":"2012-05-27T06:00:28Z","created_at":"2012-02-25T12:53:47Z","id":3544490,"html_url":"https://github.com/jacquev6/PyGithub","full_name":"jacquev6/PyGithub"}
-
+https
+GET
+api.github.com
```
Comment (sfdye) [inline tests/ReplayData/ConditionalRequestUpdate.testUpdateObjectWithoutEtag.txt]: CRLF -> LF

**Pull request merged**